static/js/hotkey.js

```diff
@@ -752,25 +752,25 @@ export function process_hotkey(e, hotkey) {
             return true;
     }
 
-    // We don't want hotkeys below this to work when recent topics is
-    // open. These involve compose box hotkeys and hotkeys that can only
-    // be done performed on a message.
-    if (recent_topics.is_visible()) {
-        return true;
-    }
-
+    const opts = {trigger: "compose_hotkey"};
     // Compose box hotkeys
     switch (event_name) {
         case "compose": // 'c': compose
-            compose_actions.start("stream", {trigger: "compose_hotkey"});
+            if (recent_topics.is_visible()) {
+                opts.stream = recent_topics.get_focused_stream();
+            }
+            compose_actions.start("stream", opts);
             return true;
         case "compose_private_message":
-            compose_actions.start("private", {trigger: "compose_hotkey"});
+            compose_actions.start("private", opts);
             return true;
         case "open_drafts":
             drafts.launch();
             return true;
         case "reply_message": // 'r': respond to message
+            if (recent_topics.is_visible()) {
+                return true;
+            }
             // Note that you can "Enter" to respond to messages as well,
             // but that is handled in process_enter_key().
             compose_actions.respond_to_message({trigger: "hotkey"});
```

Comment:
I think we should just move this one to a new block below the `recent_topics` check, rather than having extra recent_topics checks here.  Will make more sense after rebasing this.